### PR TITLE
add ADR "Service Entry Points"

### PR DESCRIPTION
**Pull request by misaka**
Had a conversation in Slack about this, again, which served as a reminder to create this ADR and update the services REAMDE.

https://ukgovernmentdfe.slack.com/archives/CGVNJ2Q2F/p1597310908061300

### CODE REVIEW

File: app/services/README.md [hunk 1]
```diff
@@ -5,8 +5,8 @@ a single place that instantiates all the services related to a model
 
 ## Conventions
 
-- Use the `execute` method as the interface on the service
-- Use named arguments to call `execute` to explicitly state what the arguments going in are
+- Use the `call` method as the interface on the service
+- Use named arguments to call `call` to explicitly state what the arguments going in are
```
Comment (gpeng) [inline app/services/README.md:9]: Is 'keyword arguments' the correct term here?

Comment (misaka) [reply app/services/README.md:9]: I think it depends on the service, I'm not sure we're dictating whether they are keyword arguments are regular arguments here.

File: docs/adr/0005-service-entry-points.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+### 2. Use "call" as an entry point
+
+Some of our service have been using `call`, and in informal polling this has
+been popular among some devs. This also mimics existing callable Ruby objects,
+which may be a good thing.
+
+## Decision
+
+Use option 2, `call`. Additionally, provider both the instance method `#call` and a class
```
Comment (gpeng) [inline docs/adr/0005-service-entry-points.md:30]: type 'provider' -> 'provide'

**Review by gpeng — CHANGES_REQUESTED** [review]

**Comment by misaka** [comment]
> Should we standardise on using this [`master`/app/services/service_pattern.rb](https://github.com/DFE-Digital/teacher-training-api/blob/master/app/services/service_pattern.rb) (and Publish [DFE-Digital/publish-teacher-training@`master`/app/services/service_pattern.rb](https://github.com/DFE-Digital/publish-teacher-training/blob/master/app/services/service_pattern.rb))? It just clears up some of the 'boiler plate' and hides the `.new` loosely enforcing the pattern.

Ya, it's a good point. We could just have an ADR about using that module, but then we might not document our choice to use `call`. Maybe it's ok to have two ADRs, I can make a new one for using the service.

File: docs/adr/0005-service-entry-points.md (hunk 2 above)
Comment (misaka) [reply docs/adr/0005-service-entry-points.md]: Fixed and squashed.

**Review by gpeng — APPROVED** [review]

**Pull request merged**